app/code/Magento/CatalogPriceDataExporter/Model/Indexer/ProductPricesFeedIndexer.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\CatalogPriceDataExporter\Model\Indexer;
+
+use Magento\CatalogPriceDataExporter\Model\EventBuilder;
+use Magento\CatalogPriceDataExporter\Model\EventPool;
+use Magento\Framework\Indexer\ActionInterface as IndexerActionInterface;
+use Magento\Framework\MessageQueue\PublisherInterface;
+use Magento\Framework\Mview\ActionInterface as MviewActionInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class responsible for indexing product prices and provide price change events
+ */
+class ProductPricesFeedIndexer implements IndexerActionInterface, MviewActionInterface
```

Comment:
It's better to create a separate model, which will collect price data instead of depending on indexer functionality. 
From indexer, we need only the "full reindex" feature.

For partial updates we may use indexer but only for the ability to use mview functionality

Reply:
Done